brozzler/js-templates/extract-outlinks.js

```diff
@@ -1,13 +1,24 @@
 // we have problems if the page has changed the definition of Set or Array
 // http://www.polyvore.com/ does this for example
 var __brzl_framesDone = new Set();
+var __brzl_popup_re = /window.open\(\s*(['"])(.*?)\1/
 var __brzl_compileOutlinks = function(frame) {
     __brzl_framesDone.add(frame);
     var outlinks = [];
     try {
         if (frame && frame.document) {
             outlinks = Array.prototype.slice.call(
                 frame.document.querySelectorAll('a[href], area[href]'));
+            popups = Array.prototype.slice.call(
+                frame.document.querySelectorAll('a[onclick]'));
+            if (popups && popups.length > 0) {
+                for (var p=0; p < popups.length; p++) {
+                    m = popups[p].onclick.toString().match(__brzl_popup_re);
```

Comment:
is there a match method for javascript regex?

maybe I'm missing something?

Reply:
You would need to make a `RegExp` obj and then call
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match

Note that the docs says: _This method is called internally in `String.prototype.match()`_

This is why I thought it may be faster to use this. You'll init `RegExp` only once outside the loop and use it repeatedly. On the contrary, when you use `string.match`, you init `RegExp` internally each time you use it.

Reply:
Thanks, Vangelis!

My quick tests are reliably ~10% faster using code based on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match

I've added this to the PR.